public/app/features/explore/QueryField.tsx

```diff
@@ -15,18 +15,18 @@ import IndentationPlugin from './slate-plugins/indentation';
 import ClipboardPlugin from './slate-plugins/clipboard';
 import RunnerPlugin from './slate-plugins/runner';
 import SuggestionsPlugin, { SuggestionsState } from './slate-plugins/suggestions';
-
 import { Typeahead } from './Typeahead';
 
 import { makeValue, SCHEMA } from '@grafana/ui';
 
-export const HIGHLIGHT_WAIT = 500;
-
 export interface QueryFieldProps {
   additionalPlugins?: Plugin[];
   cleanText?: (text: string) => string;
   disabled?: boolean;
-  initialQuery: string | null;
+  // We have both value and local state. This is usually an antipattern but we need to keep local state
+  // for perf reasons and also have outside value in for example in Explore redux that is mutable from logs
+  // creating a two way binding.
+  value: string | null;
```

Comment:
I'd like to avoid a confusion with Slate values.

Reply:
Changed to `query` mainly just wanted to remove the `initial` as it usually suggests that after initial render it should not affect the value which is not true here.